recipe/meta.yaml

```diff
@@ -49,13 +54,27 @@ requirements:
     - scikit-learn !=0.22.0
 
 test:
+  source_files:
+    - tests
   imports:
     - lightgbm
   requires:
     - pip
+    - matplotlib-base
+    - pytest
+    - cloudpickle
+    - psutil
+    # arrow
+    - pyarrow
+    - cffi
+    # panda
+    - pandas
+    # scikit-learn
     - scikit-learn
   commands:
     - pip check
+    - pytest -vv .  # [not s390x]
+    - pytest -vv -k "not (test_categorical_handle or test_categorical_handle_na or test_categorical_non_zero_inputs or test_contribs_sparse_multiclass or test_quantized_training)"  # [s390x]
```

Comment:
could you please add a comment about the particular assert failure message? Is it a numerical approximation?

Reply:
In the logs for s390x, I'm not seeing anything that is giving more details for the assertion error.

Reply:
Just kidding! Doing more reading and found it: `Not equal to tolerance rtol=1e-07, atol=0`

I'll have this added.

Reply:
those are usually safe to ignore.

Reply:
The main thing I want to know from such a comment is, why is it ok that these tests don't pass?

Reply:
I believe it's that these tests are doing some computations, and `s390x` can be not as accurate on these as test might requires. (So I've been told).

Reply:
that sounds reasonable, there's often maths errors between platforms. s390x maybe more than others? not sure.

Reply:
maybe more prone to errors because the upstream tests were probably tested by upstream on more common architectures

Reply:
I tweaked the comment a little to hopefully make it more clear.

Reply:
> maybe more prone to errors because the upstream tests were probably tested by upstream on more common architectures

sounds like a reasonable explanation to me.

Reply:
given the quantity of upstream tests I think it's ok to skip these.